### PR TITLE
update dependencies

### PR DESCRIPTION
**Pull request by cqc-melf**
# Description

Please summarise the changes.

# Related issues

Please mention any github issues addressed by this PR.

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

**Comment by cqc-melf** [comment]
I am not fully sure is the change to string is fully intentional, the old enum is still used in some parts?

File: setup.py [hunk 1]
```diff
@@ -44,10 +44,10 @@
     packages=find_namespace_packages(include=["pytket.*"]),
     include_package_data=True,
     install_requires=[
-        "pytket ~= 1.27",
-        "qiskit ~= 1.0",
+        "pytket ~= 1.28",
```
Comment (cqc-alec) [inline setup.py:47]: Can we go with the versions in #354 which are more recent?

Comment (cqc-melf) [reply setup.py:47]: Yes we can

**Review by cqc-alec — APPROVED** [review]

File: setup.py [hunk 2]
```diff
@@ -44,11 +44,11 @@
...
-        "pytket ~= 1.27",
-        "qiskit ~= 1.0",
+        "pytket ~= 1.29",
+        "qiskit ~= 1.1",
         "qiskit-algorithms ~= 0.3.0",
-        "qiskit-ibm-runtime ~= 0.23.0",
-        "qiskit-aer ~= 0.14.0",
+        "qiskit-ibm-runtime ~= 0.24.1",
+        "qiskit-aer ~= 0.14.2",
         "numpy",
```
Comment (cqc-alec) [inline setup.py:52]: Ideally we'd have "< 2" here but that's already implied by pytket's requirements, so no need.

Comment (cqc-melf) [reply setup.py]: I have added that. We might update pytket in the future, but we probably will face additional changes here.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**